rust/main/hyperlane-core/src/traits/pending_operation.rs

```diff
@@ -363,18 +365,34 @@ impl Eq for QueueOperation {}
 impl Ord for QueueOperation {
     fn cmp(&self, other: &Self) -> Ordering {
         use Ordering::*;
+
+        fn salted_hash(id: &H256, salt: &[u8]) -> H256 {
+            H256::from_slice(Keccak256::new().chain(id).chain(salt).finalize().as_slice())
+        }
+
         match (self.next_attempt_after(), other.next_attempt_after()) {
             (Some(a), Some(b)) => a.cmp(&b),
             // No time means it should come before
             (None, Some(_)) => Less,
             (Some(_), None) => Greater,
             (None, None) => {
-                if self.origin_domain_id() == other.origin_domain_id() {
-                    // Should execute in order of nonce for the same origin
-                    self.priority().cmp(&other.priority())
+                let mixing =
+                    env::var("HYPERLANE_RELAYER_MIXING_ENABLED").map_or(false, |v| v == "true");
+                if !mixing {
```

Comment:
worth adding unit tests with different values for these two environment vars

Reply:
Added unit tests